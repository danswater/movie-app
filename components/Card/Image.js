/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  position: relative;
  display: block;
  -webkit-box-flex: 0;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0;
  background: rgba(0,0,0,.05);

  img {
    display: block;
    width: 100%;
    height: auto;
    border-radius: inherit;
  }
`;

const Index = ({ alt, src }) => (
  <Container>
    <img alt={alt} src={src} />
  </Container>
);

Index.propTypes = {
  alt: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
};

export default Index;
