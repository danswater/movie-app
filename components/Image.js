import styled from 'styled-components';

const Image = styled.img`
  margin-right: 0;
  margin-bottom: 1em;
  text-align: center;
  width: 100%;
  float: ${props => (props.floatRight ? 'right' : 'none')};
  height: auto;
`;

export default Image;
