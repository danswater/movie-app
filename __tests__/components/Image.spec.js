/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Image from '../../components/Image';

describe('Components', () => {
  describe('Image', () => {
    it('should render image with default styles', () => {
      const tree = renderer.create(
        <Image
          src="https://image.tmdb.org/t/p/w500/image.jpg"
        />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule('width', '315px');
      expect(tree).toHaveStyleRule('float', 'none');
    });

    it('should render image with custom styles', () => {
      const tree = renderer.create(
        <Image
          floatRight
          width={200}
        />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule('width', '200px');
      expect(tree).toHaveStyleRule('float', 'right');
    });
  });
});
