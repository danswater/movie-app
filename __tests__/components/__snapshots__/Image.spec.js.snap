// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Image should render image with custom styles 1`] = `
.c0 {
  margin-right: 0;
  margin-bottom: 1em;
  margin-left: 1em;
  width: 200px;
  float: right;
  height: auto;
}

<img
  className="c0"
  width={200}
/>
`;

exports[`Components Image should render image with default styles 1`] = `
.c0 {
  margin-right: 0;
  margin-bottom: 1em;
  margin-left: 1em;
  width: 315px;
  float: none;
  height: auto;
}

<img
  className="c0"
  src="https://image.tmdb.org/t/p/w500/image.jpg"
/>
`;
